src/core/collector/mod.rs

```diff
@@ -203,6 +212,24 @@ impl Collector {
 		})
 	}
 
+	pub async fn map_subxt_event(&mut self, event: &SubxtEvent) -> color_eyre::Result<Vec<SubxtEvent>> {
+		match event {
+			SubxtEvent::NewHead(block_hash) => {
+				let block_hash = *block_hash;
+				let mut subxt_events = vec![SubxtEvent::NewHead(block_hash)];
+				let client = OnlineClient::<PolkadotConfig>::from_url(self.endpoint.as_str()).await?;
```

Comment:
I'm not quite sure that it is a good idea to make this call directly. We have the executor service in the collector that can use the connections pool and it can wrap all retries logic/backpressure et all. I think it is better to use it instead of opening an additional connection to an RPC node. Furthermore, `subxt` subscription provides these events for *free*, so why do we need to query them explicitly in the first place?

Reply:
I think we should move away from using the API service via channel. We need to get rid of the codebloat that comes with it. I am thinking that the API service as a global would be really neat :P

Reply:
It is not done via channels now. But there is still a logic around reconnections and backpressure that should be reused. Nevertheless, I still think that we should use events from subxt subscription and not ask a node for them one more time.